modules/pam_wheel/pam_wheel.c

```diff
@@ -199,6 +207,16 @@ perform_check (pam_handle_t *pamh, int ctrl, const char *use_group)
 
 	} else {
 	    retval = PAM_IGNORE;
+
+	    if (ctrl & PAM_AUTH_SELF && pwd->pw_uid == 0) {
+		if (ctrl & PAM_DEBUG_ARG) {
+		    pam_syslog(pamh, LOG_NOTICE, "changing authentication target to %s", fromsu);
+		}
+		if (pam_set_item(pamh, PAM_USER, fromsu) != PAM_SUCCESS) {
```

Comment:
Changing `PAM_USER` is a big hammer, its effects go far beyond the authentication.  In particular, session stack modules would use this new `PAM_USER`, for example, pam_loginuid would set loginuid to the caller uid instead of the target uid, pam_limits would apply limits for the caller uid instead of the target uid, and so on.  In other words, I don't see how to use this new feature safely.

Reply:
ok, good point. So that mean support for this kind of authentication would have to go directly into pam_unix